### DISCUSSION TITLE
2022/06/announcing-net-oss-survey-results/

### DISCUSSION THREAD

**Discussion by giscus** [SeanKilleen/seankilleen.github.io#750] [Specific Post Comments]
# 2022/06/announcing-net-oss-survey-results/

The results are in. Where do we go from here?

https://seankilleen.com/2022/06/announcing-net-oss-survey-results/

**Comment by SeanKilleen** [comment]
Getting the comments started here. Don't be shy! Feedback and suggestions, especially for the impact mapping and suggested actions, are most welcome.

**Comment by tillig** [comment]
I didn't see this survey or I'd have participated. I'm one of the active maintainers for Autofac. I'm not sure how the word was spread, but I've started finding Twitter to be an unmanageable river of input and several surveys like this only come across Twitter (at least, as I've seen). Possibly tagging/tweeting directly to well-known projects and maintainers (Xunit, AutoMapper, etc.) might help with reach.

Lots of the comments I see here resonate, so it's nice to see I'm not alone.

It may be interesting to see stats by geographic location of the active maintainers. I'm given to understand that the tax handling of OSS donations is potentially easier to deal with in the UK than it is in the US, for example, which may influence whether donations are an interesting avenue for the project. It's one of the primary reasons I'm not personally interested in donations - I don't have the wherewithal to deal with the tax side of things and deal with LLCs and non-profits and all that. I'm not a consultant in my day job, so I don't have that already set up. (That might be an interesting angle - how many of the folks interested in monetization are already independent contractors/consultants?)

It would also be interesting to see some of these stats broken down by, say, project size/popularity and maybe age. Using Autofac as an example, my impression is that, since it's been around for a while, people take it for granted, so there's no new blogging about it, no "publicity," etc. There are very few code-based contributions because (again, my impression) people assume that, due to the age/size/popularity, there's some sort of "team" dedicated to just handling everything. Autofac, right now, has the equivalent of about two and a half unpaid free-time people for the core library and all the integrations.

I'd say our biggest problem is trying to find folks who can relieve some of the burden of _ownership_. That is, I'd be happy to see a PR come in to fix a long-standing issue, but I really need folks to come in and share the weight of answering support questions, regularly participating in bug fixes and discussions, and just sort of managing the "day to day" of it. The one-off PR helps a little, but without that constant/daily support from another co-owner, it definitely limits what sort of features and fixes we can take on.

**Reply by SeanKilleen** [reply]
Hey @tillig! 👋 My thoughts below.

> I'm one of the active maintainers for Autofac

First off, thanks for that! Autofac has saved me several times, though it's been a little while since I've used it. Thanks also for maintaining a long-term "bedrock" project for the ecosystem.

> I'm not sure how the word was spread

Word was spread on Twitter and via direct messages to folks I'm connected to who are maintainers. It was also spread via the .NET Foundation's outreach channels (Twitter, Facebook) and via the .NET Foundation newsletter -- I asked them to amplify the initial survey post I left it up from January until late March to attempt to spread the word.

> Possibly tagging/tweeting directly to well-known projects and maintainers (Xunit, AutoMapper, etc.) might help with reach.

This was my first instinct. But, I polled folks I trust about that approach prior to releasing the survey and the response was overwhelmingly that I shouldn't do that because it is spammy, especially for maintainers who may receive unsolicited outreach regularly on top of trying to maintain their projects. So, I chose not to go that route out of respect for maintainers. In hindsight, I'm comfortable that I did it that way, because I think if this takes off we'll have more people opt in next time. Also, the survey was pretty long this time and this will give me a chance to refine it before running it again.

> It may be interesting to see stats by geographic location of the active maintainers
> It would also be interesting to see some of these stats broken down by, say, project size/popularity and maybe age.

I specifically didn't collect things like this because, while I agree it would be interesting, the foundational thing I wanted to capture is that all maintainers and their happiness / level of support matter to the survey and its next steps. That won out over the (admittedly interesting) insights I could obtain. I wanted to surface the varied input that could lead to a lot of solutions that might help the most maintainers, no matter their approach or project.

-----

Thanks also for your detailed feedback on your position as a maintainer! I'm happy to have it even if it's not in survey format. I'll seek to apply it to the impact map, and if you see specific places I could add items to the impact map in order to align with what you're saying, I'd love those suggestions as well. 

Thanks for taking the time to write detailed feedback!

**Comment by bijington** [comment]
Thank you for this! As @tillig mentioned a lot of the points really resonate with me also.

I don't think I have anything useful to offer in terms of suggestions but the one point I would like to raise is that the the maps have more effort still on the maintainer than other parties. This is no criticism of the research or the maps themselves but I think still highlights where some of the assistance from others could be needed. In fact I think this possibly ties into @tillig s point around getting help with the ownership burden.

I have no suggestions here but I am keen to be part of the conversation that continues to see if we can find ways to help us all out.

Good work @seankilleen
